### ISSUE TITLE
No way to get logging output from tests

### ISSUE THREAD

**Issue by Jameskmonger** [alsatian-test/alsatian#196]
Discovered this at lunch today.

```typescript
import { Test, Expect } from "alsatian";

let someFn = () => {
    console.log("whopee!");
    return 3;
};

export class LogHideTest {

    @Test()
    public shouldReturnThree() {

        // this should log whopee
        let val = someFn();

        console.log("whopay");

        Expect(val).toBe(3);

    }

}
```

I get this output:

```
> tsc && alsatian test.js

LogHideTest


shouldReturnThree
|====================|

Pass: 1/1
Fail: 0/1
Ignore: 0/1
```

How can I get my console output? It's useful for debugging.

**Comment by jamesrichford** [comment]
Does it output with `--tap` argument? What's probably happening is it's outputting be being overwritten by TAP Bark (notice the double new line after `LogHideTest`. How would you expect this to output? Above the TAP Bark output? If so maybe the solution is for TAP Bark to leave the cursor on the console above it's output?

**Comment by jamesrichford** [comment]
@Jameskmonger bump

**Comment by Jameskmonger** [comment]
@jamesrichford It works correctly with the `--tap` argument. Not sure how it should be displayed, though.

**Comment by Jameskmonger** [comment]
Tied with #127

**Comment by dshamim** [comment]
at the moment , is there anyway to get logging out from test cases onto the console for debugging etc ?(using a test runner like setup using ts-node to launch)

**Comment by jamesrichford** [comment]
If you pass the `--tap` cli argument the logging should still come out. Is that what you were looking for @dshamim? Alternatively you can use tools like visual studio code to debug your tests (which I prefer). We're looking to resolve this in all cases however.

**Comment by dshamim** [comment]
@jamesrichford Thanks for the input - 
I was able to pass the --tap cli argument and output the logs, i must have missed it earlier, as they are shown just for a second while the test is running and overridden in the final output

second way as you suggested using vs code, I'm trying to get the test cases to debug but I can't get the vs code to hit the breakpoint (I'm new to JS frameworks) I hope you can shed some light onto it or guide me in the right direction to do that ? the following is my runner.ts and launc.json

When I hit run, it runs the project and I can see the tests running, but doesn't hit any breakpoints,
would really appreciate any help in this matter to setup debugging of the test cases 

launch.json:
```
{
            "name": "ANEX.Website.ManagementPortal.Tests",
            "type": "node",
            "request": "launch",
            "runtimeExecutable": "yarn",
            "runtimeArgs": [
                "run",
                "ts-node",
                "Tests/runner.ts"
            ],
            "cwd": "${workspaceFolder}/ANEX.Website.ManagementPortal.Tests",
            "timeout": 20000,
            "protocol": "inspector",

        }

runner.ts:
```
import { TapBark } from "tap-bark";
import { TestSet, TestRunner } from "alsatian";
import { config as dotenv } from 'dotenv';

// entry point into the tests application
(async () =>
{
     // Load up any pseudo environment variables
     dotenv({ path: __dirname + '/../.env' });

    const testSet = TestSet.create();
    testSet.addTestsFromFiles('/**/*/*.spec.ts');

    const testRunner = new TestRunner();

    testRunner.outputStream
        .pipe(TapBark.create().getPipeable())
        .pipe(process.stdout);

    await testRunner.run(testSet);
})().catch(e =>
{
    console.error(e);
    process.exit(1);
});
```

```

**Comment by jamesadarich** [comment]
closed by #508
